PopPUNK/plot.py

```diff
@@ -559,15 +559,26 @@ def outputsForCytoscape(G, G_mst, isolate_names, clustering, outPrefix, epiCsv,
         save_network(G, prefix = outPrefix, suffix = suffix, use_graphml = True)
 
     # Save each component too (useful for very large graphs)
+    example_cluster_title = list(clustering.keys())[0]
     component_assignments, component_hist = gt.label_components(G)
     for component_idx in range(len(component_hist)):
-        remove_list = []
-        for vidx, v_component in enumerate(component_assignments.a):
-            if v_component != component_idx:
-                remove_list.append(vidx)
-        G.remove_vertex(remove_list)
-        G.purge_vertices()
-        save_network(G, prefix = outPrefix, suffix = "_component_" + str(component_idx + 1), use_graphml = True)
+      # Naming must reflect the full graph size
+      component_name = component_idx + 1
+      get_component_name = (use_partial_query_graph is not None)
+      # Filter the graph for the current component
+      comp_filter = G.new_vertex_property("bool")
+      for v in G.vertices():
+        comp_filter[v] = (component_assignments[v] == component_idx)
+        # If using partial query graph find the component name from the clustering
+        if get_component_name and (component_assignments[v] == component_idx):
+          example_isolate_name = seqLabels[int(v)]
+          component_name = clustering[example_cluster_title][example_isolate_name]
+          get_component_name = False
```

Comment:
just wonering why we set this to false here?

Reply:
Because clustering is defined by the components, therefore only one name is needed - after that we can avoid this bit of code